### PR TITLE
chore: dynamic shape support for clamp/min/max/floor_div/logical_and

### PR DESCRIPTION
**Pull request by keehyuna**
# Description

add dynamic shape support for clamp/min/max/floor_div/logical_and

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: tests/py/dynamo/partitioning/test_dynamic_partitioning.py [hunk 1]
```diff
@@ -53,51 +50,6 @@ def forward(self, x):
             1,
             f"Number of TRT segments should be 1 but got {trt_segments}",
         )
```
Comment (keehyuna) [inline tests/py/dynamo/partitioning/test_dynamic_partitioning.py:52]: Test case with assume_dynamic_shape_support=True flag was removed. TestDynamicPartitioning test checks if there is only trt segment in module.

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 2]
```diff
@@ -2017,8 +2017,12 @@ def aten_ops_pow(
...
 
-@dynamo_tensorrt_converter(torch.ops.aten.floor_divide.default)
-@dynamo_tensorrt_converter(torch.ops.aten.floor_divide.Scalar)
+@dynamo_tensorrt_converter(
+    torch.ops.aten.floor_divide.default, supports_dynamic_shapes=True
+)
+@dynamo_tensorrt_converter(
+    torch.ops.aten.floor_divide.Scalar, supports_dynamic_shapes=True
+)
 def aten_ops_floor_div(
```
Comment (HolyWu) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2026]: Could you also add `operator.floordiv` to the converter registry?

File: tests/py/dynamo/partitioning/test_dynamic_partitioning.py [hunk 3]
```diff
@@ -8,11 +8,8 @@
 
 from ..testing_utilities import lower_graph_testing
 
-# This testcase assumes that torch.ops.aten.clamp.default converter doesn't support
-# dynamic shapes. One should remove this testcase when the support is added.
-# This testcase tests if the graph is partitioned correctly into a TRT segment
-# and a Pytorch segment when the torch.ops.aten.clamp.default converter gets disabled
-# due to lack of dynamic shape support.
+# This testcase assumes that torch.ops.aten.clamp.default supports dynamic shapes.
```
Comment (peri044) [inline tests/py/dynamo/partitioning/test_dynamic_partitioning.py:11]: Let's remove this file completely since it is not relevant anymore

**Review by peri044 — APPROVED** [review]
LGTM. Minor comment

**Review by zewenli98 — APPROVED** [review]
LGTM

**Pull request merged**